.github/workflows/go.yml

```diff
@@ -0,0 +1,82 @@
+name: Go
+
+on: 
+  push:
+    branches:
+      - main
+      - release-*
+    paths:
+      - '**.go'
+      - 'go.mod'
+      - 'go.sum'
+      - '**/Dockerfile'
+  pull_request:
+    paths:
+      - '**.go'
+      - 'go.mod'
+      - 'go.sum'
+      - '**/Dockerfile'
+
+jobs:
+  prepare:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+      hash: ${{ steps.set-hash.outputs.hash }}
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Populate list of images to build
+      id: set-matrix
+      run: |
+        IMAGES=($(find images -mindepth 1 -maxdepth 1 -type d -exec basename {} \;))
+        IMAGES_JSON="[$(printf '"%s",' "${IMAGES[@]}")]"
+        IMAGES_JSON="${IMAGES_JSON/,\]/]}"
+        echo "matrix={\"image\":$IMAGES_JSON}" >> $GITHUB_OUTPUT
+
+    - name: Calculate a hash for images/
+      id: set-hash
+      run: echo "hash=${{ hashFiles('images/**') }}" >> $GITHUB_OUTPUT
+
+  run_tests:
```

Comment:
This job could be `fail-fast: false` so that the other image builds won't get terminated because of a build failure in another one.